olp-cpp-sdk-dataservice-read/src/generated/api/StreamApi.h

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright (C) 2020 HERE Europe B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+#pragma once
+
+#include <string>
+
+#include <boost/optional.hpp>
+#include <olp/core/client/ApiError.h>
+#include <olp/core/client/ApiNoResult.h>
+#include <olp/core/client/ApiResponse.h>
+#include <olp/dataservice/read/ConsumerProperties.h>
+#include "generated/model/SubscribeResponse.h"
+
+namespace olp {
+namespace client {
+class OlpClient;
+class CancellationContext;
+}  // namespace client
+
+namespace dataservice {
+namespace read {
+
+/**
+ * @brief The `stream` service provides the ability to consume data from a
+ * stream layer. With this service you can subscribe to a stream layer and
+ * consume messages.
+ */
+class StreamApi {
+ public:
+  using SubscribeApiResponse =
+      client::ApiResponse<model::SubscribeResponse, client::ApiError>;
+  using DeleteSubscriptionApiResponse =
```

Comment:
This seems very ugly.
I hope we had a better option here. Maybe:

```cpp
using DeleteSubscriptionApiResponse = client::ApiResponse<int, client::ApiError>;
```
Where int is the HTTP status code here and can only be 200 OK.
This is similar to CheckBlobRespone from dataservice::write::BlobApi. Please have a look and let's try to align.